src/libecalc/core/models/compressor/train/chart/variable_speed_compressor_chart.py

```diff
@@ -36,8 +36,8 @@ def _get_new_point(x: List[float], y: List[float], new_x_value) -> float:
             "Any usage of this functionality is at your own risk."
         )
         new_curves = []
-        adjust_minimum_rate_by = np.average([np.min(curve.rate) for curve in self.curves]) * control_margin
         for curve in self.curves:
+            adjust_minimum_rate_by = (np.max(curve.rate) - np.min(curve.rate)) * control_margin
```

Comment:
for single speed the above one will remain? Or is that implemented like this as well? (if implemented?)

Reply:
As far as I understand control margin is currently only implemented for variable speed compressors, am I right @olelod? It

Reply:
When this was implemented it was only done for variable speed compressor charts. It should be implemented for single speed as well at some point, but in a different task, I guess.